cardshifter-core/src/test/java/com/cardshifter/core/UserNameTest.java

```diff
@@ -0,0 +1,78 @@
+package com.cardshifter.core;
+
+import com.cardshifter.core.username.*;
+import org.junit.*;
+
+import static org.junit.Assert.fail;
+
+public class UserNameTest {
+
+    @Test
+    public void testShort() {
+        try {
+            UserName.create("a");
+        } catch (InvalidUserNameException e) {
+            fail(e.getMessage());
+        }
+    }
+
+    @Test
+    public void testLong() {
+        try {
+            UserName.create("abcdefghijklmnopqrst");
+        } catch (InvalidUserNameException e) {
```

Comment:
as you don't want the catch to happen, I would either mark the test method with throws (without using `expected = ...`, or extract the try-catch part to a method like `checkUserNameWorks("abcde...")` to avoid the repeated `try-catch` parts. Also, consider using a parametrized test.